R/get-osmdata.R

```diff
@@ -296,3 +296,23 @@ get_meta_from_cpp_output <- function (res, what = "points") {
 
     return (as.data.frame (this))
 }
+
+
+#' Extract the center matrices from `rcpp_osmdata_df` output,
+#' convert to df, and return only columns with data.
+#'
+#' The "center" components returns from `rcpp_osmdata_df()` are all named with
+#' underscore prefixes. These are prepended here with "osm_center" to provide
+#' standardised names.
+#' @noRd
+get_center_from_cpp_output <- function (res, what = "points") {
+
+    this <- res [[paste0 (what, "_center")]]
+    has_data <- apply (this, 2, function (i) any (nzchar (i)))
+    this <- this [, which (has_data), drop = FALSE]
+    if (ncol (this) > 0L) {
+        colnames (this) <- paste0 ("osm_center", colnames (this))
+    }
```

Comment:
```suggestion
    } else {
        this <- this [-seq_len (nrow (this)), ]
    }
```

Reply:
That's necessary because of the rownames. Having those means causes rows to be retained even when all columns are removed, so `nrow(this)` remains at whatever value it was, and not zero. That line removes all rows any time that the columns are removed, ensuring a null data.frame is returned.

Reply:
But we need that nrows match other data.frames, so a nrows x 0 dimensions data.frame is fine and necessary, otherwise the code will fail around https://github.com/ropensci/osmdata/blob/31b9a6e95cdd4ab0bb556f7279b73a4c18702983/R/get-osmdata-df.R#L126

And all the tests pass so I think it's fine as it is